exp/addons/internal/controllers/suite_test.go

```diff
@@ -46,6 +49,20 @@ func TestMain(m *testing.M) {
 	}
 
 	setupReconcilers := func(ctx context.Context, mgr ctrl.Manager) {
+		// Create partial cache analog to main.go.
+		partialSecretCache, err := cache.New(mgr.GetConfig(), cache.Options{
```

Comment:
Sorry very last nit: let's please integrate the changes from main.go here as well (I think should be only DefaultTransform)

Just want to make sure that we change stuff we find problems already in integration tests (could be that new functionality is only covered by integration tests and not e2e tests and then we wouldn't hit the panics)